File: client/src/app/modules/tiers/services/subscription.period.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { SubscriptionPeriodType } from '../../tiers/model/SubscriptionPeriodType';

@Injectable({
  providedIn: 'root'
})
export class SubscriptionPeriodTypeService extends AppRestService<SubscriptionPeriodType>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getSubscriptionPeriodTypes() {
    return this.getListCached(new HttpParams(), "subscription-period-types");
  }

  addOrUpdateSubscriptionPeriodType(subscriptionPeriodType: SubscriptionPeriodType) {
    this.clearListCache(new HttpParams(), "subscription-period-types");
    return this.addOrUpdate(new HttpParams(), "subscription-period-type", subscriptionPeriodType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/tiers/service/SubscriptionPeriodTypeService.java
package com.jss.osiris.modules.tiers.service;

import java.util.List;

import com.jss.osiris.modules.tiers.model.SubscriptionPeriodType;

public interface SubscriptionPeriodTypeService {
    public List<SubscriptionPeriodType> getSubscriptionPeriodTypes();

    public SubscriptionPeriodType getSubscriptionPeriodType(Integer id);
	
	 public SubscriptionPeriodType addOrUpdateSubscriptionPeriodType(SubscriptionPeriodType subscriptionPeriodType);
}
